code/modules/wiremod/components/atom/remotecam.dm

```diff
@@ -0,0 +1,448 @@
+	. = ..()
+	if(!istype(shell_parent, /obj/item/organ/internal/cyberimp/bci))
+		return
+	shell_camera = new /obj/machinery/camera (shell_parent)
+	init_camera()
+	RegisterSignal(shell_parent, COMSIG_ORGAN_IMPLANTED, PROC_REF(on_organ_implanted))
+	RegisterSignal(shell_parent, COMSIG_ORGAN_REMOVED, PROC_REF(on_organ_removed))
+	var/obj/item/organ/internal/cyberimp/bci/bci = shell_parent
+	if(bci.owner) //If somehow the camera was added while shell is already installed inside a mob, assign signals
+		if(bciuser) //This should never happen... But if it does, unassign move signal from old mob
+			UnregisterSignal(bciuser, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+		bciuser = bci.owner
+		RegisterSignal(bciuser, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+
+/obj/item/circuit_component/remotecam/bci/unregister_shell(atom/movable/shell)
+	if(shell_camera)
+		if(bciuser)
+			UnregisterSignal(bciuser, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+			bciuser = null
+		UnregisterSignal(shell_parent, list(COMSIG_ORGAN_IMPLANTED, COMSIG_ORGAN_REMOVED))
+	return ..()
+
+/obj/item/circuit_component/remotecam/bci/Destroy()
+	if(shell_camera)
+		if(bciuser)
+			UnregisterSignal(bciuser, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+			bciuser = null
+		UnregisterSignal(shell_parent, list(COMSIG_ORGAN_IMPLANTED, COMSIG_ORGAN_REMOVED))
+	return ..()
+
+/obj/item/circuit_component/remotecam/bci/proc/on_organ_implanted(datum/source, mob/living/carbon/owner)
+	SIGNAL_HANDLER
+	if(bciuser)
+		return
+	bciuser = owner
+	RegisterSignal(bciuser, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+
+/obj/item/circuit_component/remotecam/bci/proc/on_organ_removed(datum/source, mob/living/carbon/owner)
+	SIGNAL_HANDLER
+	if(!bciuser)
+		return
+	UnregisterSignal(bciuser, COMSIG_MOVABLE_MOVED, PROC_REF(update_camera_location))
+	bciuser = null
+
+/obj/item/circuit_component/remotecam/drone/register_shell(atom/movable/shell)
+	. = ..()
+	if(!istype(shell_parent, /mob/living/circuit_drone))
+		return
+	current_camera_state = FALSE //Always reset camera state for built-in shell components
+	shell_camera = new /obj/machinery/camera (shell_parent)
+	init_camera()
+
+/obj/item/circuit_component/remotecam/airlock/register_shell(atom/movable/shell)
+	. = ..()
+	if(!istype(shell_parent, /obj/machinery/door/airlock))
+		return
+	current_camera_state = FALSE //Always reset camera state for built-in shell components
+	shell_camera = new /obj/machinery/camera (shell_parent)
+	init_camera()
+
+/obj/item/circuit_component/remotecam/polaroid/register_shell(atom/movable/shell)
+	. = ..()
+	if(!istype(shell_parent, /obj/item/camera))
+		return
+	current_camera_state = FALSE //Always reset camera state for built-in shell components
+	shell_camera = new /obj/machinery/camera (shell_parent)
+	init_camera()
+
+/obj/item/circuit_component/remotecam/bci/process(seconds_per_tick)
+	if(!shell_parent || !shell_camera)
+		return
```

Comment:
Should this still be processing if there is no shell or camera?
Consider doing PROCESS_KILL and then starting the process loop again once these become valid.